ui/component/or-attribute-input/src/or-anomaly-config-chart.ts

```diff
@@ -0,0 +1,303 @@
+        }
+
+        this.datapointQuery = {
+            type: "all",
+            fromTimestamp: Date.now()- timespan * 5,
+            toTimestamp: Date.now()
+        }
+
+
+        const diffInHours = (this._endOfPeriod - this._startOfPeriod) / 1000 / 60 / 60;
+        const intervalArr = this._getInterval(diffInHours);
+
+        const stepSize: number = intervalArr[0];
+        const interval: DatapointInterval = intervalArr[1];
+
+        const lowerCaseInterval = interval.toLowerCase();
+        this._timeUnits =  lowerCaseInterval as TimeUnit;
+        this._stepSize = stepSize;
+
+        const data: ChartDataset<"line", ScatterDataPoint[]>[] = [];
+        let promises;
+
+        if(this.dataProvider) {
+            await this.dataProvider(this._startOfPeriod, this._endOfPeriod, (interval.toString() as TimeUnit), stepSize).then((dataset) => {
+                dataset.forEach((set) => { data.push(set); });
+            });
+        } else {
+            promises = this.assetAttributes.map(async ([assetIndex, attribute], index) => {
+
+                const asset = this.assets[assetIndex];
+                const descriptors = AssetModelUtil.getAttributeAndValueDescriptors(asset.type, attribute.name, attribute);
+                const label = Util.getAttributeLabel(attribute, descriptors[0], asset.type, false);
+                const unit = Util.resolveUnits(Util.getAttributeUnits(attribute, descriptors[0], asset.type));
+                const colourIndex = index % this.colors.length;
+
+
+                //limits anomaly data
+                let datasets = await this.getAnomalyLimits(asset,attribute,this.datapointQuery)
+                    datasets = datasets as ChartDataset<"line",ScatterDataPoint[]>[];
+                    if(datasets.length !== 0){
+                        let dataset = datasets[2];
+                        (dataset as any).assetId = asset.id;
+                        (dataset as any).attrName = attribute.name;
+                        (dataset as any).unit = unit;
+                        dataset = datasets[0];
+                        data.push(dataset);
+                        dataset = datasets[1];
+                        data.push(dataset);
+                        for (let i = 2; i < datasets.length; i=i+2){
+                            dataset = datasets[i];
+                            data.push(dataset);
+                            dataset = datasets[i+1];
+                            data.push(dataset);
+                        }
+                    }
+            });
+        }
+
+        if(promises) {
+            await Promise.all(promises);
+        }
+        if(data[0] && data[0].data.length !== 0){
+            this._data = data;
+        }
+        this._loading = false;
+
+    }
+
+    protected async getAnomalyLimits(asset: Asset, attribute:Attribute<any>,query:AssetDatapointQueryUnion): Promise<ChartDataset<"line", ScatterDataPoint[]>[]>{
```

Comment:
Same for this function in regards to function separation.

Reply:
50% of the length of this functions is variable declaration for the styling of the graph. The other half could be compacted but is expanded for readability.